rootfs/etc/nginx/lua/configuration.lua

```diff
@@ -26,11 +41,16 @@ function _M.call()
     return
   end
 
-  ngx.req.read_body()
+  local backends = fetch_request_body()
+  if not backends then
+    ngx.log(ngx.ERR, "dynamic-configuration: POST carries empty response body")
+    ngx.status = ngx.HTTP_BAD_REQUEST
+    return
+  end
```

Comment:
This is a change of behaviour - i can not think of anything bad with it, but also don't see why it it necessary. Did you add it merely so that you can assert for absence of this logs in the e2e test?

Reply:
The intention with this is to not update the config if the request_body is `nil`, as we were doing previously. This could be problematic, and is essentially a silent failure mode - since there are no symptoms i.e. 200 was being returned. 

Understandably, if the balancer module cannot parse the backends data, then it doesn't sync the update - so the failure is somewhat limited, but I still think that the logging and 5xx response increases visibility here.